src/bio/bio_context.c

```diff
@@ -609,15 +610,17 @@ bio_ioctxt_close(struct bio_io_context *ctxt)
 	}
 
 	rc = bio_bs_hold(xs_ctxt->bxc_blobstore);
-	if (rc)
-		return rc;
+	is_hold = rc == 0;
 
-	rc = bio_blob_close(ctxt, false);
+	if (is_hold)
+		rc = bio_blob_close(ctxt, false);
 
 	/* Free the io context no matter if close succeeded */
 	d_list_del_init(&ctxt->bic_link);
 	D_FREE(ctxt);
-	bio_bs_unhold(xs_ctxt->bxc_blobstore);
+
```

Comment:
@NiuYawei is this the right approach here?   Obviously, we want to clean up the memory at some point

Reply:
Ok. As workaround will free memory in upper layer. Otherwise this will be a memory leak.